### PR TITLE
Create contribution guide

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
- Make compilation process more generic to make external contributors' lifes easier.
- Create a contribution guide describing the main aspects to get started.

Solves https://github.com/kinvolk/inspektor-gadget/issues/31.
Solves https://github.com/kinvolk/inspektor-gadget/issues/60.

### CODE REVIEW

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -28,9 +28,9 @@ jobs:
       id: publish-registry
       uses: elgohr/Publish-Docker-Github-Action@2.8
       with:
-        name: docker.io/kinvolk/gadget
+        name: ${{ secrets.DOCKER_REPO }}
         username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (alban) [inline .github/workflows/inspektor-gadget.yml:32]: `DOCKER_USERNAME` -> `CONTAINER_REGISTRY_USERNAME`

(and other occurrences below)

File: .github/workflows/inspektor-gadget.yml [hunk 2]
```diff
@@ -28,9 +28,9 @@ jobs:
       id: publish-registry
       uses: elgohr/Publish-Docker-Github-Action@2.8
       with:
-        name: docker.io/kinvolk/gadget
+        name: ${{ secrets.DOCKER_REPO }}
```
Comment (alban) [inline .github/workflows/inspektor-gadget.yml:31]: Line 41, near `make kubectl-gadget`:
```
      env:
        CONTAINER_REPO: ${{ secrets.DOCKER_REPO }}
```

Maybe the same for the other `make` invokations.

File: .github/workflows/inspektor-gadget.yml (hunk 1 above)
Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml]: Yes, something went wrong during a rebase.

File: .github/workflows/inspektor-gadget.yml (hunk 2 above)
Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml]: Done.

File: .github/workflows/inspektor-gadget.yml [hunk 3]
```diff
@@ -65,26 +72,29 @@ jobs:
...
+        IMAGE_TAG: ${{ steps.publish-registry.outputs.snapshot-tag }}
       run: |
-        export IMAGE_TAG=${{steps.publish-registry.outputs.snapshot-tag}}
         echo "Using IMAGE_TAG=$IMAGE_TAG"
 
-        make -C integration build push
-
         TESTS_DOCKER_ARGS="-e KUBECONFIG=/root/.kube/config -v /home/runner/.kube:/root/.kube -v /home/runner/work/_temp/.minikube:/home/runner/work/_temp/.minikube" \
-            make -C integration test
+            make -C integration build test
```
Comment (alban) [inline .github/workflows/inspektor-gadget.yml:86]: We need to take care whether `make -C integration build` builds the container image in docker on the host or in minikube.

If it is meant to be built on the host, the Kubernetes cluster in minikube will not have access to the container image unless it is pushed to the registry.

If it is meant to be built in minikube, I guess you need to add the following either here or in the Makefile:
```
`eval $(shell $(MINIKUBE) docker-env | grep =)`
```
(there is an example of that in `gadget-container/Makefile`).

And then, change `imagePullPolicy: Always` to `Never` in `integration/gadget-integration-tests-job.yaml`.

Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml:86]: I'm not getting it. We are not deploying `integration/gadget-integration-tests-job.yaml` here, we are just using docker and passing the minikube kubeconfig to it. The Github actions appear to be working fine at this point:
- On this repo : https://github.com/kinvolk/inspektor-gadget/runs/762231976?check_suite_focus=true
- On my personal fork: https://github.com/mauriciovasquezbernal/inspektor-gadget/runs/762218463?check_suite_focus=true

Comment (alban) [reply .github/workflows/inspektor-gadget.yml:86]: You're right. Sorry, I got things mixed up.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+```
+$ kubectl gadget deploy | kubectl apply -f -
+```
+* Make changes in the traceloop repository and compile with `make`
+* Generate the new gadget image and deploy it to minikube:
+```
+$ make -C gadget-container/ minikube
+```
+
+Note that the minikube image only works with the traceloop gadget.
```
Comment (alban) [inline CONTRIBUTING.md:101]: This should no longer be true after https://github.com/kinvolk/inspektor-gadget/pull/113

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+## Testing
+
+### Unit tests
+
+You can run the different unit tests by:
+
+```
+make test
```
Comment (alban) [inline CONTRIBUTING.md:110]: For consistency with other commands, add a `$ `.
```suggestion
$ make test
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+$ make -C gadget-container/ minikube
+```
+
+Note that the minikube image only works with the traceloop gadget.
+
+## Testing
+
+### Unit tests
+
+You can run the different unit tests by:
```
Comment (alban) [inline CONTRIBUTING.md:107]:
```suggestion
You can run the different unit tests with:
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+```
+
+### Integration tests
+
+The integration tests use a Kubernetes cluster to deploy and test Inspektor Gadget.
+Be sure that you have a valid kubeconfig and run:
+
+```
+export KUBECONFIG=... # not needed if valid config in $HOME/.kube/config
+make integration-tests
```
Comment (alban) [inline CONTRIBUTING.md:120]:
```suggestion
$ export KUBECONFIG=... # not needed if valid config in $HOME/.kube/config
$ make integration-tests
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  - [Integration tests](#integration-tests)
+- [Authoring PRs](#authoring-prs)
+- [Good first issues](#good-first-issues)
+- [Proposing new features](#proposing-new-features)
+- [Porting BCC gadgets](#porting-bcc-gadgets)
+
+## Architecture
+
+It's highly recommended to read the [architecture](./Documentation/architecture.md) documentation before starting
+to play with Inspektor Gadget.
```
Comment (joaquimrocha) [inline CONTRIBUTING.md:24]: Nitpick: I would have chosen sometihng else more serious than `play` here, but I'm no fun. :slightly_smiling_face:

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+## Setup developer environment
+
+- [Fork](https://github.com/kinvolk/inspektor-gadget/fork) and clone this repo.
+    - `git clone git@github.com:your_account/inspektor-gadget.git`.
+- Install [Docker](https://docs.docker.com/get-docker/), [Golang](https://golang.org/doc/install).
```
Comment (joaquimrocha) [inline CONTRIBUTING.md:30]:
```suggestion
- Install [Docker](https://docs.docker.com/get-docker/) and [Golang](https://golang.org/doc/install).
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+installing the compilers to build it.
+
+## Workflows
+
+### Github Actions
+
+This repo uses Github actions as CI. It compiles and uploads the Inspektor Gadget
+executable and gadget container image. It also runs unit and some integration tests.
+A fork of this project should
+enable them in the repo settings page and add the following
```
Comment (joaquimrocha) [inline CONTRIBUTING.md:76]: Nitpick: This line is broken earlier than normal.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+ * Consider reviewing the changes yourself before opening a PR. It is likely
+   you will catch errors when looking at your code critically and thus save the
+   reviewers (and yourself) time.
+
+ * Use the PR's description as a "cover letter" and give the context you think
+   reviewers might need. Use the PR's description to explain why you are
+   proposing the change, give an overview, raise questions about yet-unresolved
+   issues in your PR, list TODO items etc.
+
+PRs which follow these rules are easier to review and merge.
```
Comment (joaquimrocha) [inline CONTRIBUTING.md:156]: We really need to add this stuff to the `contribute` repo. Just a comment to say we may change this part later and just point to the contribute repo.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+
+## Workflows
+
+### Github Actions
+
+This repo uses Github actions as CI. It compiles and uploads the Inspektor Gadget
+executable and gadget container image. It also runs unit and some integration tests.
+A fork of this project should
+enable them in the repo settings page and add the following
+[secrets](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository) to be able to use them:
```
Comment (alban) [inline CONTRIBUTING.md:77]:
```suggestion
A fork of this project should enable them in the repo settings page and add the
following
[secrets](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
to be able to use them:

```

**Pull request merged**